### PR TITLE
Drop Centos7 support

### PR DESCRIPTION
**Pull request by NvTimLiu**
To fix https://github.com/rapidsai/cudf/issues/15583

We plan to drop CentOS 7 (which uses glibc 2.17) RAPIDS 24.06. The [java/ci/Dockerfile.centos7](https://github.com/rapidsai/cudf/blob/branch-24.06/java/ci/Dockerfile.centos7), refor or to :

https://docs.rapids.ai/notices/rsn0037/

https://github.com/rapidsai/build-planning/issues/23

Change to run build in Rocky8 docker container

## Checklist
- [X] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [X] The documentation is up to date with these changes.

### CODE REVIEW

File: java/ci/Dockerfile.rocky [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+ARG CUDA_VERSION=11.8.0
+ARG OS_RELEASE=8
+ARG TARGETPLATFORM=linux/amd64
+# multi-platform build with: docker buildx build --platform linux/arm64,linux/amd64 <ARGS> on either amd64 or arm64 host
+# check available offcial arm-based docker images at https://hub.docker.com/r/nvidia/cuda/tags (OS/ARCH)
+FROM --platform=$TARGETPLATFORM nvidia/cuda:$CUDA_VERSION-devel-rockylinux$OS_RELEASE
+ARG TOOLSET_VERSION=11
+### Install basic requirements
+RUN dnf install -y scl-utils gcc-toolset-${TOOLSET_VERSION}
+RUN dnf --enablerepo=powertools install -y git zlib-devel maven tar wget patch ninja-build
```
Comment (jlowe) [inline java/ci/Dockerfile.rocky:32]: Curious if these need to be on separate lines or could be combined to remove the need for another layer in the image?

File: java/ci/Dockerfile.rocky [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+## pre-create the CMAKE_INSTALL_PREFIX folder, set writable by any user for Jenkins
+RUN mkdir /usr/local/rapids && mkdir /rapids && chmod 777 /usr/local/rapids && chmod 777 /rapids
```
Comment (jlowe) [inline java/ci/Dockerfile.rocky:34]: Nit:
```suggestion
RUN mkdir /usr/local/rapids /rapids && chmod 777 /usr/local/rapids /rapids
```

**Review by jlowe — APPROVED** [review]
Minor nits otherwise lgtm.

File: build.sh [hunk 3]
```diff
@@ -110,7 +110,7 @@ function buildAll {
 
 function buildLibCudfJniInDocker {
     local cudaVersion="11.5.0"
-    local imageName="cudf-build:${cudaVersion}-devel-centos7"
+    local imageName="cudf-build:${cudaVersion}-devel-rockey8"
```
Comment (bdice) [inline build.sh:113]:
```suggestion
    local imageName="cudf-build:${cudaVersion}-devel-rocky8"
```

**Review by bdice — COMMENTED** [review]
One typo -- otherwise LGTM.

File: build.sh [hunk 4]
```diff
@@ -110,7 +110,7 @@ function buildAll {
 
 function buildLibCudfJniInDocker {
     local cudaVersion="11.5.0"
```
Comment (bdice) [inline build.sh:112]: Should this use 11.8?

File: java/ci/Dockerfile.rocky (hunk 2 above)
Comment (NvTimLiu) [reply java/ci/Dockerfile.rocky]: Good suggestion!

File: build.sh (hunk 3 above)
Comment (NvTimLiu) [reply build.sh]: Good catch, thanks!

File: java/ci/Dockerfile.rocky (hunk 1 above)
Comment (NvTimLiu) [reply java/ci/Dockerfile.rocky]: Good suggestion, put these 2 lines together to reduce another layer in the dockerimage.

File: java/ci/Dockerfile.rocky [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+   scl enable gcc-toolset-${TOOLSET_VERSION} \
+      "cmake .. \
+         -DCMAKE_BUILD_TYPE=Release \
+         -DZSTD_FROM_INTERNET=ON \
+         -DREDIS_STORAGE_BACKEND=OFF && \
+      cmake --build . --parallel 4 --target install" && \
+   cd ../.. && \
+   rm -rf ccache-${CCACHE_VERSION}
+
+# disable cuda container constraints to allow running w/ elder drivers on data-center GPUs
```
Comment (bdice) [inline java/ci/Dockerfile.rocky:61]:
```suggestion
# disable cuda container constraints to allow running w/ older drivers on data-center GPUs
```

**Review by bdice — APPROVED** [review]

File: build.sh (hunk 4 above)
Comment (NvTimLiu) [reply build.sh]: Good catch, updated, thanks!

**Comment by NvTimLiu** [comment]
@rapidsai/ops-codeowners  can you help review? Thanks!

**Comment by bdice** [comment]
I merged the upstream and resolved a merge conflict from #15619.

**Review by raydouglass — COMMENTED** [review]
Removing `ops-codeowners` from the required reviews since it doesn't seem there are any file changes that we're responsible for. Feel free to add us back if necessary.

**Review by raydouglass — APPROVED** [review]
Did it out of order, so approved to remove any blockers.

**Comment by bdice** [comment]
Thanks @NvTimLiu!

**Pull request merged**